File: hangman.py
#!/usr/bin/env python3
import random
import string

# String to list
def stol(str):
	result = []
	for c in str:
		result.append(c)
	return result

# Print list as string
def printl(arr):
	for c in arr:
		print(c, end='')
	print()

with open('wordlist.txt', 'r') as wordlist:
	words = wordlist.readlines()

for word in words:
	words[words.index(word)] = word.replace('\n', '')

word = words[random.randint(0, len(words) - 1)]
shown = '_' * len(word)
guesses = 7
enum = list(enumerate(word))
guessed = []

word = stol(word)
shown = stol(shown)

print('Guess a letter:')
while (guesses > 0 and shown != word):
	printl(shown)
	try:
		guess = input().lower()
	except KeyboardInterrupt:
		exit()

	if (guess not in string.ascii_lowercase or len(guess) != 1):
		print('Not a letter')
		continue

	if (guess in guessed):
		print('Already guessed')
		continue

	guessed.append(guess)

	if (guess in word):
		for tup in enum:
			if (tup[1] == guess):
				shown[tup[0]] = word[tup[0]]
	else:
		print('Not in word')
		guesses -= 1
		print(f'{guesses} guesses left')

print("The word was: ")
printl(word)

File: hangman.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <cmath>
using namespace std;

int main()
{
	string letters = "abcdefghijklmnopqrstuvwxyz";

	// Get wordlist
	ifstream file("wordlist.txt");
	string line;
	vector<string> words;
	while (getline(file, line))
	{
		words.push_back(line);
	}
	file.close();

	// Get random word from wordlist
	srand((unsigned int)time(NULL));
	string ans = words[rand() % 100 + 1];

	// Shown initialization
	char shown[ans.size() + 1];
	for (int i = 0; i < ans.size(); i++)
	{
		shown[i] = '_';
	}

	// Main loop
	string in;
	int guesses = 7;
	vector<char> guessed;

	cout << "Guess a letter: \n";
	while (shown != ans && guesses > 0)
	{
		// Guess prompt
		cout << shown << "\n";
		cin >> in;
		
		// Check if guess is character
		char guess = tolower(in[0]);
		if (letters.find(guess) == string::npos)
		{
			cout << "Not a letter\n";
			continue;
		}
		
		// Check if already guessed
		if (count(guessed.begin(), guessed.end(), guess))
		{
			cout << "Already guessed\n";
			continue;
		}

		// Check if guess is in ans
		if (ans.find(guess) != string::npos)
		{
			// In
			for (int i = 0; i < ans.length(); i++)
			{
				if (ans[i] == guess)
				{
					shown[i] = guess;
				}
			}
		}
		else
		{
			guesses--;
			cout << "Not in word" << "\n" << guesses << " guesses left\n";
		}

		// Add to already guessed
		guessed.push_back(guess);
	}
	cout << "The word was: " << ans << "\n";
	return 0;
}
